vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/ModifyStylePage.java

```diff
@@ -0,0 +1,63 @@
+package com.vaadin.flow.component.map;
+
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.map.configuration.Coordinate;
+import com.vaadin.flow.component.map.configuration.feature.MarkerFeature;
+import com.vaadin.flow.component.map.configuration.style.Icon;
+import com.vaadin.flow.router.Route;
+
+@Route("vaadin-map/modify-style")
+public class ModifyStylePage extends Div {
+    public ModifyStylePage() {
+        Map map = new Map();
+
+        MarkerFeature marker = new MarkerFeature(new Coordinate(0, 0),
+                createIcon());
+        map.getFeatureLayer().addFeature(marker);
+
+        NativeButton setImage = new NativeButton("Set image", e -> {
+            Icon icon = createIcon();
+            marker.getStyle().setImage(icon);
+        });
+        setImage.setId("set-style-image");
+
+        NativeButton setImageScale = new NativeButton("Set image scale",
+                e -> marker.getStyle().getImage().setScale(5));
+        setImageScale.setId("set-image-scale");
+
+        Span renderCount = new Span("0");
+        renderCount.setId("render-count");
+        // Register event listener for the rendercomplete event, and increase
+        // render count
+        //@formatter:off
+        renderCount.getElement().executeJs(
+                "const map = $0;" +
+                        "map.configuration.on('rendercomplete', () => {" +
+                        "  const nextValue = parseInt(this.textContent) + 1;" +
+                        "  this.textContent = nextValue.toString();" +
+                        "});",
+                map.getElement()
+        );
```

Comment:
I am assuming that this test is aiming to verify that even if `forceRender` is called multiple times, the debouncing mechanism makes the render to be called only once (in most cases). However, on my tests removing the debouncing control from `forceRender` doesn't change the number of renders, so it seems that OL has itself some control to avoid multiple renders.
I think it's still important to have the debounce, since it avoids looping through the layers multiple times unnecessarily.

Reply:
The intention was actually to check that the map rendered at all after updating styles, which was not the case previously. Testing that this results in a single render is more of a side-effect. Another interesting thing is that rendering the map does not necessarily mean that a layer was re-rendered, these only re-render if they have been marked as changed. I added another assertion to the test that tries to verify that by checking that the layer's revision increased (which is was `layer.changed()` does).

Regarding the debounce: Yes, OpenLayers does some debouncing on its own, but I couldn't find any clear documentation on it. So I thought adding our own debounce as a safe-guard would make sense.